store/src/java/com/zimbra/cs/service/util/S3GlobalConfigUtil.java

```diff
@@ -0,0 +1,481 @@
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.ldap.LdapUtil;
+import com.zimbra.soap.admin.message.AddS3ConfigRequest;
+import com.zimbra.soap.admin.message.DeleteS3ConfigRequest;
+import com.zimbra.soap.admin.type.Attr;
+import com.zimbra.soap.type.GlobalExternalStoreConfig;
+import com.zimbra.soap.type.GlobalS3BucketConfiguration;
+
+public class S3GlobalConfigUtil {
+
+	public static final String ACTIVE = "ACTIVE";
+	public static final String DELETED = "DELETED";
+	public static final String PROV_AMAZON = "AmazonS3";
+	public static final String PROV_OPENIO = "OpenIO";
+	public static final String PROV_CEPH = "Ceph";
+	public static final String PROV_NETAPP = "NetApp";
+	public static final String HTTPS = "https";
+	public static final String HTTP = "http";
+	public static final int OK = 200;
+	
+
+	/**
+     * Parse the json which comes from the GlobalExternalStoreConfig LDAP parameter to object.
+     *
+     * @param gescLdapJson
+     * @return GlobalExternalStoreConfig
+     * @throws ServiceException
+     */
+    public static GlobalExternalStoreConfig parseJsonToGesc(String gescLdapJson) throws ServiceException {
+    	
+    	GlobalExternalStoreConfig gesc = new GlobalExternalStoreConfig();
+    	
+    	try {
+    		
+    		if(gescLdapJson != null && !gescLdapJson.isEmpty()) {
+    			JSONObject gescLdapJO = new JSONObject(gescLdapJson);
+    			JSONArray gs3bucketsJA = gescLdapJO.getJSONArray(AdminConstants.E_GLOBAL_S3_BUCKET_CONFIGURATIONS);
+    			if(gs3bucketsJA != null) {
+	    			for(int i = 0; i < gs3bucketsJA.length(); i++) {
+	    				JSONObject gs3bucketConfJO = gs3bucketsJA.getJSONObject(i);
+	    				boolean bucketIsActive = ((String) gs3bucketConfJO.get(AdminConstants.A_S3_BUCKET_STATUS)).equalsIgnoreCase("ACTIVE");
+	    				if(bucketIsActive) {
+	    					GlobalS3BucketConfiguration gs3bucketConf = new GlobalS3BucketConfiguration();
+	    					for (Iterator it = gs3bucketConfJO.keys(); it.hasNext(); ) {
+	    						Object fieldObject = it.next();
+	    						if(fieldObject instanceof String) {
+	    							String field = String.valueOf(fieldObject);
+	    							String value = (String) gs3bucketConfJO.get(field);
+	    							setValueByReflection(gs3bucketConf, field, value);
+	    						}
+	    					}
+	    					gesc.addGlobalS3BucketConfiguration(gs3bucketConf);
+	    				}
+	    			}
+    			}
+    		}
+		
+		} catch (JSONException e) {
+			ZimbraLog.mailbox.warn("failed to parse ldap global_external_store_config parameter", e);
+		}
+		
+		return gesc;
+    	
+    }
+    
+    /**
+     * Parse the Attr list inside of request element to a GlobalS3BucketConfiguration object
+     *
+     * @param request
+     * @return GlobalS3BucketConfiguration
+     * @throws ServiceException
+     */
+    public static GlobalS3BucketConfiguration parseRequestToGs3bc(Object request) throws ServiceException {
+    	
+    	GlobalS3BucketConfiguration gs3bc = new GlobalS3BucketConfiguration();
```

Comment:
Lets keep separate methods for delete and Add

Reply:
done!